configuration/redisnative/redis.go

```diff
@@ -0,0 +1,300 @@
+		if err != nil {
+			return m, fmt.Errorf("redis store error: can't parse maxRetryBackoff field: %s", err)
+		}
+		m.maxRetryBackoff = time.Duration(parsedVal)
+	}
+
+	if val, ok := meta.Properties[failover]; ok && val != "" {
+		failover, err := strconv.ParseBool(val)
+		if err != nil {
+			return m, fmt.Errorf("redis store error: can't parse failover field: %s", err)
+		}
+		m.failover = failover
+	}
+
+	// set the sentinelMasterName only with failover == true.
+	if m.failover {
+		if val, ok := meta.Properties[sentinelMasterName]; ok && val != "" {
+			m.sentinelMasterName = val
+		} else {
+			return m, errors.New("redis store error: missing sentinelMasterName")
+		}
+	}
+
+	return m, nil
+}
+
+// Init does metadata and connection parsing.
+func (r *ConfigurationStore) Init(metadata configuration.Metadata) error {
+	m, err := parseRedisMetadata(metadata)
+	if err != nil {
+		return err
+	}
+	r.metadata = m
+
+	if r.metadata.failover {
+		r.client = r.newFailoverClient(m)
+	} else {
+		r.client = r.newClient(m)
+	}
+
+	if _, err = r.client.Ping().Result(); err != nil {
+		return fmt.Errorf("redis store: error connecting to redis at %s: %s", m.host, err)
+	}
+
+	r.replicas, err = r.getConnectedSlaves()
+
+	return err
+}
+
+func (r *ConfigurationStore) newClient(m metadata) *redis.Client {
+	opts := &redis.Options{
+		Addr:            m.host,
+		Password:        m.password,
+		DB:              defaultDB,
+		MaxRetries:      m.maxRetries,
+		MaxRetryBackoff: m.maxRetryBackoff,
+	}
+
+	// tell the linter to skip a check here.
+	/* #nosec */
+	if m.enableTLS {
+		opts.TLSConfig = &tls.Config{
+			InsecureSkipVerify: m.enableTLS,
+		}
+	}
+
+	return redis.NewClient(opts)
+}
+
+func (r *ConfigurationStore) newFailoverClient(m metadata) *redis.Client {
+	opts := &redis.FailoverOptions{
+		MasterName:      r.metadata.sentinelMasterName,
+		SentinelAddrs:   []string{r.metadata.host},
+		DB:              defaultDB,
+		MaxRetries:      m.maxRetries,
+		MaxRetryBackoff: m.maxRetryBackoff,
+	}
+
+	/* #nosec */
+	if m.enableTLS {
+		opts.TLSConfig = &tls.Config{
+			InsecureSkipVerify: m.enableTLS,
+		}
+	}
+
+	return redis.NewFailoverClient(opts)
+}
+
+func (r *ConfigurationStore) getConnectedSlaves() (int, error) {
+	res, err := r.client.DoContext(context.Background(), "INFO", "replication").Result()
+	if err != nil {
+		return 0, err
+	}
+
+	// Response example: https://redis.io/commands/info#return-value
+	// # Replication\r\nrole:master\r\nconnected_slaves:1\r\n
+	s, _ := strconv.Unquote(fmt.Sprintf("%q", res))
+	if len(s) == 0 {
+		return 0, nil
+	}
+
+	return r.parseConnectedSlaves(s), nil
+}
+
+func (r *ConfigurationStore) parseConnectedSlaves(res string) int {
+	infos := strings.Split(res, infoReplicationDelimiter)
+	for _, info := range infos {
+		if strings.Contains(info, connectedSlavesReplicas) {
+			parsedReplicas, _ := strconv.ParseUint(info[len(connectedSlavesReplicas):], 10, 32)
```

Comment:
Could we skip this error safely?

Reply:
This is refered from https://github.com/dapr/components-contrib/blob/master/state/redis/redis.go#L162